### DISCUSSION TITLE
@graphql-eslint/alphabetize support autofix ?

### DISCUSSION THREAD

**Discussion by neviaumi** [graphql-hive/graphql-eslint#935] [Ideas]
https://github.com/dotansimha/graphql-eslint/blob/master/packages/plugin/src/rules/alphabetize.ts

It possible to make rules `@graphql-eslint/alphabetize` support autofix? 

My case was i have 2000+ line graphql schema right now. manual fix it and apply the rules is terrible tasks.

Or some online tools can help me for initial setup ?

**Comment by neviaumi** [comment]
turnout i find my solution there

https://github.com/graphql/graphql-js/issues/2036


And according to here, seem `@graphql-eslint/alphabetize` behaviour would difference to graphql utilities provided
https://github.com/cjoudrey/graphql-schema-linter/commit/0e32302630c0af36f5a51081ac8141558aeaa15e

**Comment by dimaMachina** [comment]
Sure, I will add autofix but only for cases where there are no comments between or around fields. And I'll change the order to `lexicographic`.
Nice catch!

**Comment by dimaMachina** [comment]
@davidNHK available in `3.8.0-alpha-db02c77.0` version, could you try it and give your feedback?

**Reply by neviaumi** [reply]
> only for cases where there are no comments between or around fields


i said the original issue on graphql/graphql-js#2036

the code below can handle comment as well

```js
const { lexicographicSortSchema,  buildSchema, printSchema } = require('graphql')
const fs = require('fs')
const schema = fs.readFileSync('./schema.graphql', { encoding: 'utf-8'})
const schemaV2 = buildSchema(schema)
printSchema(lexicographicSortSchema(schemaV2))
```

**Reply by dimaMachina** [reply]
I adjusted autofix to handle comments in `3.8.0-alpha-2d2b247.0` version

**Comment by dotansimha** [comment]
Fixed in `@graphql-eslint/eslint-plugin@3.8.0` 🎉

**Comment by neviaumi** [comment]
I tested with my production schema now 

And open 1 issue (Not a big deal )
https://github.com/B2o5T/graphql-eslint/issues/955

One more suggestion would 

Given
```
input Z {}
input A {}
input C {}

query {
 .....
}
```

It only able sort the field under Z,A,C and query

the input order didn't get sorted. it would perfect if input / type can sorted as well

**Reply by neviaumi** [reply]
Rules i configured , may be i just get wrong

```
 {     '@graphql-eslint/alphabetize': [
        'error',
        {
          arguments: [
            'FieldDefinition',
            'Field',
            'DirectiveDefinition',
            'Directive',
          ],
          fields: [
            'ObjectTypeDefinition',
            'InterfaceTypeDefinition',
            'InputObjectTypeDefinition',
          ],
          selections: ['OperationDefinition', 'FragmentDefinition'],
          values: ['EnumTypeDefinition'],
          variables: ['OperationDefinition'],
        },
      ],
}
```

**Reply by dimaMachina** [reply]
Hi @davidNHK. Currently it’s possible to sort only fields, but I can improve that in the next version. How do you think which configuration options will be the best for sorting definitions as well?

**Reply by neviaumi** [reply]
I think for the configuration, according to

https://spec.graphql.org/October2021/#sec-Documen

how about have extra config attributes named `document` ? 
value i feel like just follow current conversion (array of string) and possible value only [Definition](https://spec.graphql.org/October2021/#Definition) is enough ? 


so my updated config may be look like that 

```
 {     '@graphql-eslint/alphabetize': [
        'error',
        {
          arguments: [
            'FieldDefinition',
            'Field',
            'DirectiveDefinition',
            'Directive',
          ],
          fields: [
            'ObjectTypeDefinition',
            'InterfaceTypeDefinition',
            'InputObjectTypeDefinition',
          ],
          selections: ['OperationDefinition', 'FragmentDefinition'],
          values: ['EnumTypeDefinition'],
          variables: ['OperationDefinition'],
          document: ['Definition']
        },
      ],
}
```

**Reply by dimaMachina** [reply]
@davidNHK available in [@graphql-eslint/eslint-plugin@3.9.0](https://github.com/B2o5T/graphql-eslint/releases/tag/%40graphql-eslint%2Feslint-plugin%403.9.0) 🚀

**Reply by gaving** [reply]
Thanks for this, was a bit confused how to order the definitions until I added `definitions: true` into my custom alphabetize rule.
